### PR TITLE
feat: remove app mode config from admin screen

### PR DESCRIPTION
**Pull request by eventualbuddha**
Instead, we fetch app  mode from `/machine-config` which pulls machine ID from `VX_MACHINE_ID` and app mode from `VX_APP_MODE` environment variables.

This will, for now, make it harder to switch between app modes in development. You'll have to restart the server with the right environment variable and run it like so:

```
$ VX_APP_MODE=VxPrint yarn start
```

### CODE REVIEW

File: src/AppRoot.tsx [hunk 1]
```diff
@@ -181,7 +175,7 @@ class AppRoot extends React.Component<Props, State> {
     isFetchingElection: false,
     isLiveMode: false,
     isPollsOpen: false,
-    machineId: '---',
+    machineConfig: { appMode: VxMarkOnly, machineId: '0000' },
```
Comment (beausmith) [inline src/AppRoot.tsx]: nit: Drop one zero here so all the `productId`'s are three digits?

File: src/config/types.ts [hunk 2]
```diff
@@ -15,18 +15,30 @@ export interface Provider<T> {
...
+  isVxMark: true,
+  isVxPrint: false,
+} as const
+export const VxMarkPlusVxPrint = {
   name: 'VxMark + VxPrint',
   isVxPrint: true,
   isVxMark: true,
+} as const
+export type AppMode =
+  | typeof VxMarkOnly
```
Comment (beausmith) [inline src/config/types.ts]: Q: should this line start with a pipe?

File: src/utils/machineConfig.ts [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+  Provider,
+  MachineConfig,
+  getAppMode,
+  AppModeNames,
+} from '../config/types'
+import fetchJSON from './fetchJSON'
+
+interface MachineConfigResponse {
+  machineId: string
+  appMode: string
```
Comment (beausmith) [inline src/utils/machineConfig.ts]: Should `appMode` be more tightly typed?

File: src/AppRoot.tsx (hunk 1 above)
Comment (eventualbuddha) [reply src/AppRoot.tsx]: @benadida says this is typically 4  ‾\_(ツ)_/‾

File: src/config/types.ts (hunk 2 above)
Comment (eventualbuddha) [reply src/config/types.ts]: That's what prettier did. It's just a way to visually make it easier to grok.

File: src/AppRoot.tsx (hunk 1 above)
Comment (beausmith) [reply src/AppRoot.tsx]: LOLz… what'evs. 😄

File: src/config/types.ts (hunk 2 above)
Comment (beausmith) [reply src/config/types.ts]: Cool. Good to know this syntax is acceptable. I like.

**Pull request merged**